distributed/tests/test_failed_workers.py

```diff
@@ -417,14 +417,19 @@ async def test_restart_timeout_on_long_running_task(c, s, a):
 
 @gen_cluster(client=True, scheduler_kwargs={"worker_ttl": "500ms"})
 async def test_worker_time_to_live(c, s, a, b):
+    from distributed.scheduler import heartbeat_interval
+
+    # worker removal is also controlled by 10 * heartbeat
     assert set(s.workers) == {a.address, b.address}
+    interval = 10 * heartbeat_interval(len(s.workers)) + 0.5
+
     a.periodic_callbacks["heartbeat"].stop()
     await asyncio.sleep(0.010)
     assert set(s.workers) == {a.address, b.address}
 
     start = time()
     while set(s.workers) == {a.address, b.address}:
-        await asyncio.sleep(0.050)
-        assert time() < start + 2
+        await asyncio.sleep(interval)
```

Comment:
How long does this test take to run?  If it's longer than a second can we mark it with `pytest.mark.slow`.  If it's longer than 10 seconds then we might want to find some other way around this.

Reply:
This takes ~7 seconds. 
> real    0m7.363s

Happy to make as slow if that's close enough to 10